src/clr-angular/forms/common/if-error.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Directive, Input, Optional, TemplateRef, ViewContainerRef} from "@angular/core";
+import {Subscription} from "rxjs";
+
+import {ControlService} from "./providers/control.service";
+
+@Directive({selector: "[clrIfError]"})
+export class ClrIfError {
+    constructor(@Optional() private service: ControlService, private template: TemplateRef<any>,
+                private container: ViewContainerRef) {
+        if (!this.service) {
+            throw new Error(
+                "clrIfError can only be used within a form control container element like clr-input-container");
+        } else {
+            this.displayError(false);
+        }
+    }
+
+    @Input("clrIfError") error: string;
+
+    private subscription: Subscription;
+    private state: boolean = false;
```

Comment:
Nitpick: after reading the code, I think `state` is too generic of a name for this property. I'd rather have `displayed`, or `currentError`, or anything that indicates this property tracks whether we are already displaying the error at the moment or not.
As I said it's a nitpick, nothing I feel strongly about or that I consider critical.